pkg/operator/vtctld/deployment.go

```diff
@@ -93,6 +94,9 @@ func UpdateDeploymentImmediate(obj *appsv1.Deployment, spec *Spec) {
 	// Set labels on the Deployment object.
 	update.Labels(&obj.Labels, spec.Labels)
 
+	// Set user labels on the Deployment object.
```

Comment:
Move this logic into `UpdateDeployment`, this is a change that we want to put behind a rollout and isn't necessarily safe to change immediately. Additionally, similarly to what I suggested with the vtgate deployment logic, we only need to ask our deployment to set `UserLabels` on the pods, not the deployment object.